homeassistant/components/auth/login_flow.py

```diff
@@ -151,8 +153,13 @@ def __init__(self, flow_mgr):
         else:
             handler = data['handler']
 
+        if handler[0] == 'trusted_networks':
```

Comment:
This is not good.

By doing this, we're starting to conflate concerns of the auth provider with the generic abstraction. It means the abstraction is not good and that needs to be changed. 

I don't know right now how we should evolve it, maybe we need to decouple the manager from the flow, allowing the manager to pass in an extra auth context.

Reply:
This also comes with having to decouple the HTTP views from using the helper.

And all these things should happen in separate PRs. One thing at a time.

Reply:
Part 3 of #15700 was implement login's own flow manager.